### ISSUE TITLE
Save user country country

### ISSUE THREAD

**Issue by alexfauquette** [openfoodfacts/hunger-games#887]
Followup of #879 (@teolemon, @benbenben2  if you have an opinion on the behavior)

In `/settings` we allow to set the language but not the country.

![image](https://github.com/openfoodfacts/hunger-games/assets/45398769/a7d42020-385b-43b2-84a3-e37664c8fa45)

The idea would be to to save in the localeStorage the user country.

By default country should not be defined (equivalent of being set to world)

If a default country is set, it should be used in most games. Especially in the eco-score dashboard

@Gaurav-Verma07 you might be interested by this follow up

**Comment by teolemon** [comment]
A drop-down with world and all countries could be interesting, 

If we're looking for more sophistication, a multi select, or even groupings of Spanish/French/English speaking countries could be interesting as well.

**Comment by Gaurav-Verma07** [comment]
@alexfauquette 
So, we've to add a coutry drop-down in settings page and save that country in the localStorage.
Additionally, we use this country as default in eco-score page .

**Comment by Gaurav-Verma07** [comment]
Hey @alexfauquette any updates on this feature.
I see my PR closed.

**Comment by alexfauquette** [comment]
Seems you closed it 🤔

![image](https://github.com/openfoodfacts/hunger-games/assets/45398769/81a7efee-8d96-49ad-8504-cf364fbcd05d)

> should I use /assets/countries.json or /pages/sco-score/index/countryNames to populate countries dropdown in settings.

Nice observation. Might be better to use `/assets/countries.json` Otherwise you might get people from other countries asking to add their one.

The eco-score is a bit special. It's a project we launched mostly for france and neighbors county so we used a restricted list of country.

I see two solution to avoid the bug. Modify the country selector in eco-score page to support all countries, or do a hack to add the URL country in the list of options
